elastic/metricbeat-tests-poc
# Support for verification of the Helm charts using BDD approach

## What is this PR doing?
It creates the first feature file and the implementation of the steps for verifying that the Elastic's Helm charts for Metricbeat are correct.

For that, each step in the feature file will represent each of the verification points required for the Helm chart.

The implementation of each step will leverage the local binaries for `helm` and `kubectl`, so each step will run different commands for these tools reading the output and checking those values.

The test cluster will be managed by [kind](https://kind.sigs.k8s.io/), and the binaries needed for it (kind, kubectl, helm) are installed by the shell script that runs the e2e tests.

## Why is it important?
We are going to manually verify the Helm charts, but adding this automated test suite will support the verification of the charts with any frequency: with a PR, on each release, on feature freeze, etc.

## How to test this PR?

1. Please check that you have `helm` and `kubectl` installed in your local machine.
1. From the `e2e` directory, run:
```shell
$ godog -t helm
```

## Related issues
- Closes #99
